### PR TITLE
Update release process

### PR DESCRIPTION
**Pull request by LilianBoulard**

### CODE REVIEW

File: RELEASE_PROCESS.md [hunk 1]
```diff
@@ -33,14 +32,17 @@ here are the main steps and appropriate resources:
...
+    with the command `pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://test.pypi.org/simple/ dirty_cat`
+12. Finally, if that works fine, you can push to the official package index with
     `twine upload dist/*`
-12. It is also good practice at this point to create a new environment
-    and try installing and using the library.
-13. To finish the procedure, create a new release on the GitHub repository.
+13. It is also good practice at this point to create a new environment
+    and try installing and using the library (for example by launching examples).
+    Warning: be aware that conda & such cache packages, so doing `pip install dirty_cat`
+    will use the cached version and not the new one
```
Comment (jovan-stojanovic) [inline RELEASE_PROCESS.md:44]: Or use `pip install dirty-cat==package_version` to avoid confusion

**Review by jovan-stojanovic — CHANGES_REQUESTED** [review]
Another PR also change the RELEASE_PROCESS file #350

File: RELEASE_PROCESS.md [hunk 2]
```diff
@@ -33,14 +32,18 @@ here are the main steps and appropriate resources:
...
    `twine upload dist/* --repository-url https://test.pypi.org/legacy/`
    This is useful to test if the build done by twine and the push to the
-   package indexer is working (if you can pip install dirty_cat from
-   test.pypi)
-10. Install the new release from the test package index on a dedicated environment
-11. Finally, if that works fine, you can push to the official package index with
+   package indexer is working
+10. Create a new Python environment and install the package from the test PyPi
+11. Install the new release from the test package index on a dedicated environment
+    with the command `pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://test.pypi.org/simple/ dirty_cat`
```
Comment (jovan-stojanovic) [inline RELEASE_PROCESS.md:38]: 10 and 11 are repetitive, maybe you can merge them

**Review by jovan-stojanovic — CHANGES_REQUESTED** [review]
The rest is good.

**Review by jovan-stojanovic — APPROVED** [review]

**Review by GaelVaroquaux — APPROVED** [review]
LGTM. Merging

**Pull request merged**